lib/ibmdb.js

```diff
@@ -190,73 +189,79 @@ IBMDB.prototype.escapeName = function(name) {
   return '"' + name + '"';
 };
 
-
 /**
  * Execute the sql statement
  *
  */
 IBMDB.prototype.executeSQL = function(sql, params, options, callback) {
   debug('DB2.prototype.executeSQL (enter)',
         sql, params, options);
-  var self = this;
-  var conn = self.connection;
-  var stmt = {};
 
-  if (!self.connection.connected) {
-    self.dataSource.connected = false;
-    self.dataSource.connecting = false;
-  }
-  stmt.noResults = (options && options.noResults) ? options.noResults : false;
+  var self = this;
 
-  if (options.transaction) {
-    conn = options.transaction.connection;
-  }
-
-  var limit = 0;
-  var offset = 0;
-  // This is standard DB2 syntax. LIMIT and OFFSET
-  // are configured off by default. Enable these to
-  // leverage LIMIT and OFFSET.
-  if (!this.useLimitOffset) {
-    var res = sql.match(self.limitRE);
-    if (res) {
-      limit = Number(res[1]);
-      sql = sql.replace(self.limitRE, '');
-    }
-    res = sql.match(this.offsetRE);
-    if (res) {
-      offset = Number(res[1]);
-      sql = sql.replace(self.offsetRE, '');
+  // var executeStatement = function(conn, sql, params, options, callback) {
+  var executeStatement = function(conn) {
+    // var self = this;
+    var limit = 0;
+    var offset = 0;
+    var stmt = {};
+
+    stmt.noResults = (options && options.noResults) ? options.noResults : false;
+    // This is standard DB2 syntax. LIMIT and OFFSET
+    // are configured off by default. Enable these to
+    // leverage LIMIT and OFFSET.
+    if (!self.useLimitOffset) {
+      var res = sql.match(self.limitRE);
+      if (res) {
+        limit = Number(res[1]);
```

Comment:
I would personally use `parseInt` here and I would use `Number()` where I want to check if the given string completely represents a numeric value, float or integer.

One main difference between `parseInt()` Vs `Number()`:
`parseInt()` parses up to the first non-digit and, whereas `Number()` wants to convert the entire string into a number.

```
> var foo = '123foo';
undefined
> parseInt(foo);
123
> Number(foo);
NaN
```